### PR TITLE
Unified JS code style

### LINKED ISSUE #404: JavaScript code linting and formatting
We should have standards for how JavaScript code is linted and formatted so it is homogeneous across files.

Copy the tools and settings from the openst repository that @pgev established.

Include auto-formatting of all existing files in this ticket.

Change indentation to `2` spaces.
Change line width hard wrap limit max character to `79` spaces.

### PR DESCRIPTION
**Pull request by schemar**
Code was formatted using `eslint`.

We have around 300 linting issues left, which I cannot fix all in
this PR.
I suggest we start fixing the files one at a time when we work in them.

Fixes #404

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -42,7 +45,8 @@
     "ganache": "sh tools/runGanacheCli.sh",
     "build-package": "node tools/build_package.js",
     "prepare": "npm run compile-all && npm run build-package",
-    "deploy:gateway": "npm run compile && node tools/blue_deployment/scripts/step1_origin_contracts.js"
+    "deploy:gateway": "npm run compile && node tools/blue_deployment/scripts/step1_origin_contracts.js",
+    "lint": "eslint {migrations,test,test_integration,tools} -c .eslintrc.json --ext js"
```
Comment (sarvesh-ost) [inline package.json]: maybe add a pre-commit hook to run this.  Refer [this.](https://github.com/OpenSTFoundation/mosaic.js/blob/bcb49479763f782b35cf447c4fc893e16a1a6384/package.json#L63)

Comment (schemar) [reply package.json]: It still returns more than 700 issues. Not sure a pre-commit hook helps 🤔

**Comment by schemar** [comment]
> There are conflicts, please resolve them.

There will be more conflicts after #633 will be merged. I will resolve them in one go then 👍

**Comment by schemar** [comment]
Remember to set ES6 to true

**Comment by schemar** [comment]
Integration tests fail 😱 I will check it tomorrow.

File: test/lib/gatewaylib/hash_redeem_intent.js [hunk 2]
```diff
@@ -20,25 +20,23 @@
 
 const GatewayLib = artifacts.require('GatewayLib');
 
-const Utils = require('../../gateway/eip20_cogateway/helpers/co_gateway_utils');
 const BN = require('bn.js');
+const Utils = require('../../gateway/eip20_cogateway/helpers/co_gateway_utils');
```
Comment (sarvesh-ost) [inline test/lib/gatewaylib/hash_redeem_intent.js]: why did you changed the sequence?

File: test/lib/gatewaylib/prove_account.js [hunk 3]
```diff
@@ -24,13 +24,22 @@ const StubData = require('../../data/proof.json');
...
-    encodedAccountValue,
-    path,
-    stateRoot,
-    expectedStorageRoot;
+
+  let accountProof;
+
+
+  let encodedAccountValue;
+
```
Comment (sarvesh-ost) [inline test/lib/gatewaylib/prove_account.js]: Remove extra lines.

File: test/lib/messagebus/messagebus_utils.js [hunk 4]
```diff
@@ -19,483 +19,454 @@
...
+    const intentHash = web3.utils.soliditySha3({
+      type: 'bytes32',
+      value: 'intent',
+    });
+    const nonce = 1;
+    const gasPrice = 0x12a05f200;
+    const sender = accounts[7];
+    const gasLimit = 0;
+    const gasConsumed = 0;
+
```
Comment (sarvesh-ost) [inline test/lib/messagebus/messagebus_utils.js]: Extra line?

File: test/lib/organization/constructor.js [hunk 5]
```diff
@@ -158,32 +141,22 @@ contract('Organization.constructor()', async (accounts) => {
...
-      ),
+      Organization.new(owner, admin, workers, expirationHeight),
       'Worker address cannot be null\\.',
     );
   });
 
-  it('should not deploy if the expiration height is not in the future for ' +
-    'any workers',
+  it(
+    'should not deploy if the expiration height is not in the future for '
```
Comment (sarvesh-ost) [inline test/lib/organization/constructor.js]: Should start from above line.

File: test/test_lib/constants.js [hunk 6]
```diff
@@ -1,6 +1,5 @@
-
 const constants = {
-  //For truffle tests, we do not have separate chains, so keeping block time as 2 for both constants.
+  // For truffle tests, we do not have separate chains, so keeping block time as 2 for both constants.
```
Comment (sarvesh-ost) [inline test/test_lib/constants.js]: We should probably delete this file.

File: test/test_lib/event_decoder.js [hunk 7]
```diff
@@ -1,29 +1,30 @@
...
-  getFormattedEvents: function (eventsData) {
-    var formattedEvents = {};
+      const currEventName = currEvent.name ? currEvent.name : currEvent.event;
 
-    for (var i = 0; i < eventsData.length; i++) {
-      var currEvent = eventsData[i]
-        , currEventName = currEvent.name ? currEvent.name : currEvent.event
-        , currEventAddr = currEvent.address
-        , currEventParams = currEvent.events ? currEvent.events : currEvent.args;
+      const currEventAddr = currEvent.address;
```
Comment (sarvesh-ost) [inline test/test_lib/event_decoder.js]: I think these are unintentional new line.

File: test_integration/lib/anchor.js [hunk 8]
```diff
@@ -26,7 +26,7 @@ const EventDecoder = require('../../test/test_lib/event_decoder');
  * This class represents anchor contract on a chain.
  */
 class Anchor {
-    /**
+  /**
```
Comment (sarvesh-ost) [inline test_integration/lib/anchor.js]:
```suggestion
    /**
```

File: tools/deployment_tool/address_generators.js [hunk 9]
```diff
@@ -27,30 +27,30 @@ const rlp = require('rlp');
...
      */
-    constructor(startAddress = '0x0000000000000000000000000000000000010000') {
-        this.nextAvailableAddress = startAddress;
-    }
+  constructor(startAddress = '0x0000000000000000000000000000000000010000') {
+    this.nextAvailableAddress = startAddress;
+  }
 
-    /**
+  /**
```
Comment (sarvesh-ost) [inline tools/deployment_tool/address_generators.js]:
```suggestion
    /**
```
Also at other places in this file.

File: test/lib/gatewaylib/hash_redeem_intent.js (hunk 2 above)
Comment (schemar) [reply test/lib/gatewaylib/hash_redeem_intent.js]: according to eslint, node module requires must be before relative requires.

**Pull request merged**